### DISCUSSION TITLE
Improve feedback when saving changes to bracket configuration in the tournament client

### DISCUSSION THREAD

**Discussion by ILW8** [ppy/osu#19094] [Ideas] [answered]
1) In response to a request for feedback from running the lazer tournament client for [The Roundtable](https://osu.ppy.sh/wiki/en/Tournaments/The_Roundtable)
____
Perhaps the single most noticeable inconvenience with the tournament client was with saving settings. The behavior has been quite inconsistent and the lack of feedback when saving changes makes it difficult to know whether a change has been successfully saved or not. 
For example, when editing the bracket, it seems that each bracket node has to be saved individually (it seems that  you can't change multiple nodes and then save them all at once). This isn't obvious as there's no feedback when clicking the save button. I would expect the save button to gray out once all changes have been saved, and for the save button to reactivate whenever a change that needs to be saved happens.

That being said, it's entirely possible some of this comes down to user error. However, I think work could be done to make the process of setting up a tournament more intuitive and obvious.

**Comment by peppy** [comment]
The save button should save all changes globally, so if that's not happening could you please provide a specific example (a video showing this occurring should be enough). The expected UX is that you should be able to make any changes anywhere in the tournament client, and only save when you're happy with what you've done (to avoid accidentally breaking the bracket you're using on an ongoing basis when testing new configurations etc.).

I can definitely agree with the save button becoming disabled when there are no pending changes.

**Reply by ILW8** [reply]
Sure, I will try reproducing this in a moment.

**Reply by ILW8** [reply]
I have only been able to reproduce this issue with a node's match time. Perhaps I'm misremembering and the issue only affects match time saving...
[https://jame.btmc.live/Desktop%202022.07.13%20-%2008.25.36.07.mp4](https://jame.btmc.live/Desktop%202022.07.13%20-%2008.25.36.07.mp4)

**Reply by peppy** [reply]
The issue there is that you didn't "commit" the textbox by hitting enter or clicking away on the first edit. This should probably be done automatically when switching to a new panel for editing, i'll look into it.

**Comment by peppy** [comment, marked as answer]
The save button feedback portion should be addressed via #19100

**Reply by peppy** [reply]
Date commit issue should be addressed via https://github.com/ppy/osu/pull/19102.
